mood-analysis.rb

```diff
@@ -28,5 +34,62 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+# print out date + mood by calling analyze mood
+def week_moods(array)
+  array.each do |day|
+    date = day[0..4]
+    puts "#{date} #{analyze_mood(day)}"
+  end
+end
+
+def happy_days(array)
+  happy = 0
+  days = 0
+  array.each do |day|
+    mood = analyze_mood(day)[-3..-1]
+    if mood == ":-)"
+      happy += 1
+    end
+    days += 1
+    if happy == 3
+      break
+    end
+  end  
+  if happy == 3
+    puts "It takes #{days} entries for 3 happy days to occur"
+  else
+    puts "You are having quite a bad week."
+  end
+end
+
+# returns most common mood
+def overall_mood(array)
+  mood_array = []
+  happy = 0
+  sad = 0
+  neutral = 0
+  array.each do |day|
+    mood = analyze_mood(day)
+    mood_array << mood
+  end
+  mood_array.each do |mood|
+    happy += 1 if mood == ":-)"
+    sad += 1 if mood == ":-("
+    neutral += 1 if mood == ":-|"
+  end
+  if happy == [happy, sad, neutral].max
+    puts "The most common mood is :-)"
+  elsif sad == [happy, sad, neutral].max
+    puts "The most common mood is :-("
+  else
+    puts "The most common mood is :-|"
+  end
+end
```

Comment:
This looks like it works, but is not the easiest approach. What was your reasoning for using an array? --> That's why I wanted the README. Again, this works, but can be simplified significantly using a hash. Hashes are often used for situations like this, where we want to count the number of occurances of something, especially in whiteboarding problems.